### ISSUE TITLE
Documentation genneration

### ISSUE THREAD

**Issue by jasontatton** [Concurnas/Concurnas#6]
Concurnas could benefit from a documentation generation system, just like javadoc

**Comment by DeflatedPickle** [comment]
I don't think it needs a whole new system, Javadoc works fine. But a system compatible with Javadoc, but with extended syntax might be nice.
Unless the new system would work differently, like using raw syntax or annotations to generate the documentation.
